addons/mail/sfu_server/src/services/ws.js

```diff
@@ -0,0 +1,140 @@
+import { WebSocketServer } from "ws";
+
+import * as config from "#src/config.js";
+import { WS_CLOSE_CODE } from "#src/shared/enums.js";
+import { Logger, extractRemoteAddress } from "#src/utils/utils.js";
+import { SESSION_CLOSE_CODE } from "#src/models/session.js";
+import { Bus } from "#src/shared/bus.js";
+import { verify, AuthenticationError } from "#src/services/auth.js";
+import { Channel } from "#src/models/channel.js";
+
+const logger = new Logger("WS");
+/** @type {Map<number, import("ws").WebSocket>} */
+const unauthenticatedWebSockets = new Map();
+const authenticatedWebSockets = new Set();
+let pendingId = 0;
+/** @type {import("ws").WebSocketServer} */
+let server;
+
+/**
+ * @param {Parameters<import("ws").WebSocketServer>[0]} options WebSocketServer options
+ * @returns {Promise<import("ws").WebSocketServer>}
+ */
+export async function start(options) {
+    logger.info("starting...");
+    server = new WebSocketServer(options);
+    /**
+     * This is the entry point for clients, from here the client initiates a webSocket connection to the server,
+     * provides the secret/jwt for authentication, and if the authentication is successful, the server creates a session
+     * and connects it, otherwise the server closes the webSocket (and no rtc session is created).
+     */
+    server.on("connection", (webSocket, req) => {
```

Comment:
Hum, I don't get how we reach this point. It runs on the same HTTP port, so is there a namespacing of the route? Or it is from the port used?

Reply:
yes that's it, since it runs on the http server, the only difference is `http(s)` vs `ws(s)`